src/Symfony/Bridge/Doctrine/Form/ChoiceList/IdReader.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\Doctrine\Form\ChoiceList;
```

Comment:
I think that doesn't really make sense. ChoiceList is very form specific (values are always strings, names must follow the naming conventions of forms, ...). Making ChoiceList more generic would just make maintenance harder, and I'd like to avoid that - maintenance of the Form component is hard enough as it is.